ui/components/src/FormatControls/FormatControls.tsx

```diff
@@ -21,21 +21,26 @@ import {
 } from '@perses-dev/core';
 import { ReactElement } from 'react';
 import { OptionsEditorControl } from '../OptionsEditorLayout';
-import { SettingsAutocomplete } from '../SettingsAutocomplete';
+import { SettingsAutocomplete, SettingsAutocompleteOption } from '../SettingsAutocomplete';
 
 export interface FormatControlsProps {
   value: FormatOptions;
   onChange: (unit: FormatOptions) => void;
   disabled?: boolean;
 }
 
-type AutocompleteUnitOption = UnitConfig & { id: FormatOptions['unit'] };
+type AutocompleteUnitOption = Omit<SettingsAutocompleteOption, 'id'> &
+  UnitConfig & {
+    id: NonNullable<FormatOptions['unit']>;
+  };
 
-const KIND_OPTIONS: AutocompleteUnitOption[] = Object.entries(UNIT_CONFIG)
+const KIND_OPTIONS: readonly AutocompleteUnitOption[] = Object.entries(UNIT_CONFIG)
   .map(([id, config]) => {
+    const unit = id as NonNullable<FormatOptions['unit']>;
```

Comment:
All FormatOptions have a value, so from typescript view this is never null
```suggestion
    const unit = id as FormatOptions['unit'];
```